### PR TITLE
Update the satinfo and convinfo files so the v16.3 parallel experimen…

### PR DESCRIPTION
**Pull request by emilyhcliu**
The gfsv16.3 parallel testing begins from 20211015 18Z.
The relevant GSI fix files (convinfo & satinfo) were modify to include the observations to be assimilated operationally in real-time.  A corresponding change in the global-work under **/parm/config.anal** is necessary so the proper fix files will be used in the assimilation based on cycle date/time.  

In **GSI**:
(1) a subdirectory "gfsv16.3" has been created under /gsi.fd/fix/gfsv16_historical/  for gfsv16.3 implementation specifically.  
The change can be checked on WCOSS-2 in the following directory:
/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow/gfs.v16.3.0/sorc/gsi.fd/fix/gfsv16_historical/gfsv16.3

(2) The corresponding changes for the convinfo and satinfo files will be made in the GSI repository (gfsda.v16.3 release branch) --- [GSI PR # 412](https://github.com/NOAA-EMC/GSI/pull/412)

In **global-workflow**:
(1) The /parm/config.anal is modified for exporting the updated convinfo and satinfo files from GSI.
(2) The changes can be viewed [here](https://github.com/NOAA-EMC/global-workflow/pull/876/files)

The related issue is[ issue #877](https://github.com/NOAA-EMC/global-workflow/issues/877)

### CODE REVIEW

**Comment by emilyhcliu** [comment]
@KateFriedman-NOAA Please add Russ to review this PR.  Thanks.

**Review by RussTreadon-NOAA — COMMENTED** [review]
I'm not the right person to review these changes since I don't know what observations we want to assimilate or monitor in the GFS v16.3 retrospective parallel.   I was not directly involved in the implementations which turned on/off the indicated observations.  The fix files changed in config.anal do not exist in the g-w repo.   These files files come from the fix submodule associated with the NOAA-EMC/GSI repo.   As such, we can not confirm that the comments in config.anal indicating the nature of the fix file change are actually in the given fix file.

**Comment by emilyhcliu** [comment]
@RussTreadon-NOAA I already asked Andrew and Haixia to review the related GSI PR ([#412](https://github.com/NOAA-EMC/GSI/pull/412))  I will them to review this one.  

@KateFriedman-NOAA Could you add @HaixiaLiu-NOAA and @ADCollard to review this PR.  Thanks.

File: parm/config/config.anal [hunk 1]
```diff
@@ -117,12 +117,12 @@ if [[ $RUN_ENVIR == "emc" ]]; then
 
 #   Assimilate DO-2 GeoOptics
     if [[ "$CDATE" -ge "2021031712" && "$CDATE" -lt "2021091612" ]]; then
-        export CONVINFO=$FIXgsi/gfsv16_historical/global_convinfo.txt.2021031712
+        export CONVINFO=$FIXgsi/gfsv16_historical/gfsv16.3/global_convinfo.txt.2021031712
```
Comment (RussTreadon-NOAA) [inline parm/config/config.anal]: Dogwood directory `/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow/gfs.v16.3.0/sorc/gsi.fd/fix/gfsv16_historical/gfsv16.3` does not contain `global_convinfo.txt.2021031712`.   This directory contains `global_convinfo.txt.2021091612` and `global_convinfo.txt.2022031612`.

File: parm/config/config.anal [hunk 2]
```diff
@@ -159,12 +159,12 @@ if [[ $RUN_ENVIR == "emc" ]]; then
 
 #   Turn off assimilation of S-NPP CrIS
     if [[ "$CDATE" -ge "2021052118" && "$CDATE" -lt "2021092206" ]]; then
-	export SATINFO=$FIXgsi/gfsv16_historical/global_satinfo.txt.2021052118
+	export SATINFO=$FIXgsi/gfsv16_historical/gfsv16.3/global_satinfo.txt.2021052118
```
Comment (RussTreadon-NOAA) [inline parm/config/config.anal]: As above, Dogwood directory `/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow/gfs.v16.3.0/sorc/gsi.fd/fix/gfsv16_historical/gfsv16.3` does not contain `global_satinfo.txt.2021052118`.   It contains `global_satinfo.txt.2021092206` and `global_satinfo.txt.2021102612`.

**Comment by RussTreadon-NOAA** [comment]
Am I looking in the right directory?

```
russ.treadon@dlogin04:/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow/gfs.v16.3.0/sorc/gsi.fd/fix/gfsv16_historical/gfsv16.3> ls -l
total 908
-rwxr-xr-x 1 emily.liu da  34087 Jun 22 21:54 global_convinfo.txt.2021091612
-rwxr-xr-x 1 emily.liu da  34087 Jun 22 21:56 global_convinfo.txt.2022031612
-rw-r--r-- 1 emily.liu da 423313 Jun 22 21:12 global_satinfo.txt.2021092206
-rw-r--r-- 1 emily.liu da 423313 Jun 22 21:12 global_satinfo.txt.2021102612
-rw-r--r-- 1 emily.liu da    405 Jun 22 22:09 README
```

**Comment by emilyhcliu** [comment]
> Am I looking in the right directory?
> 
> ```
> russ.treadon@dlogin04:/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow/gfs.v16.3.0/sorc/gsi.fd/fix/gfsv16_historical/gfsv16.3> ls -l
> total 908
> -rwxr-xr-x 1 emily.liu da  34087 Jun 22 21:54 global_convinfo.txt.2021091612
> -rwxr-xr-x 1 emily.liu da  34087 Jun 22 21:56 global_convinfo.txt.2022031612
> -rw-r--r-- 1 emily.liu da 423313 Jun 22 21:12 global_satinfo.txt.2021092206
> -rw-r--r-- 1 emily.liu da 423313 Jun 22 21:12 global_satinfo.txt.2021102612
> -rw-r--r-- 1 emily.liu da    405 Jun 22 22:09 README
> ```

@RussTreadon-NOAA Yes.  That's the right directory.
The changes from this PR are testing on WCOSS-2 with rocoto workflow.  It works expected.  
EXPDIR: /lfs/h2/emc/da/noscrub/emily.liu/para_gfs/v163testc
ROTDIR: /lfs/h2/emc/ptmp/emily.liu/comrot/v163testc

**Comment by RussTreadon-NOAA** [comment]
I do not see files `global_convinfo.txt.2021031712` and `global_satinfo.txt.2021052118` in the directory indicated by the updated `config.anal`.  According to the updated `config.anal`, these two files should be in `$FIXgsi/gfsv16_historical/gfsv16.3`.   

Looking at log files in your `ROTDIR` this gets expanded to `/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow/gfs.v16.3.0/fix/fix_gsi/gfsv16_historical/gfsv16.3`.    This directory does not contain a 2021031712 convinfo file.   It does not contain a 2021052118 satinfo file.   

Files `global_convinfo.txt.2021031712` and `global_satinfo.txt.2021052118` exist one directory up in `/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow/gfs.v16.3.0/fix/fix_gsi/gfsv16_historical/`

Your test covers 2021101518 to 2021103000.   This period will not exercise lines in question in `config.anal`

Your test is OK.   Lines 120 and 162 of the modified `config.anal` are not OK.

**Comment by emilyhcliu** [comment]
@RussTreadon-NOAA Let me check....I probably did something stupid.

**Comment by emilyhcliu** [comment]
> I do not see files `global_convinfo.txt.2021031712` and `global_satinfo.txt.2021052118` in the directory indicated by the updated `config.anal`. According to the updated `config.anal`, these two files should be in `$FIXgsi/gfsv16_historical/gfsv16.3`.
> 
> Looking at log files in your `ROTDIR` this gets expanded to `/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow/gfs.v16.3.0/fix/fix_gsi/gfsv16_historical/gfsv16.3`. This directory does not contain a 2021031712 convinfo file. It does not contain a 2021052118 satinfo file.
> 
> Files `global_convinfo.txt.2021031712` and `global_satinfo.txt.2021052118` exist one directory up in `/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow/gfs.v16.3.0/fix/fix_gsi/gfsv16_historical/`
> 
> Your test covers 2021101518 to 2021103000. This period will not exercise lines in question in `config.anal`
> 
> Your test is OK. Lines 120 and 162 of the modified `config.anal` are not OK.

@RussTreadon-NOAA Thanks for checking.  The convinfo and satinfo files under /gfsv16_historical/gfsv16.3 are correct.
It is the modifications that I made in the config.anal were wrong.  I should not have gfsv16.3 in the path for 2021031712 convinfo and 202105218 satinfo.  

Luckily, these stupid mistakes did not impact the parallel experiment.

**Comment by emilyhcliu** [comment]
For GPSRO, update nsig_ext from 56 to 45.  The original value (56) is suitable for 64 model layers.  The updated value (45) is more suitable for 127 model layer.  
The change has been validated by @LidiaCucurull-NOAA and approved by @ADCollard

**Comment by KateFriedman-NOAA** [comment]
@emilyhcliu I'll submit this PR when @ADCollard approves it or you tell me it's ready to go. Waiting on confirmation. Thanks!

**Comment by aerorahul** [comment]
@ADCollard 
Are we good to merge these changes?

**Review by ADCollard — APPROVED** [review]
All good.  Sorry for the delay.

**Pull request merged**